### PR TITLE
Separate test error into separate component

### LINKED ISSUE #3456: Add clickable docs links to error messages
### Current behavior:
Does not support markdown

### Desired behavior:
Should support markdown for links to docs.

Originally this issue was to add full Markdown support, but that'll require a bit more work at a later date. This PR will just add links and the scaffolding to more easily add markdown support later on. 

- [x] Add Markdown rendering that only linkifys `on.cypress.io` domains

### PR DESCRIPTION
**Pull request by lilaconlee**
- fixes #3456 
- Addresses #3762

### CODE REVIEW

**Comment by lilaconlee** [comment]
Ok, updated a handful of errors to get a sense of the different content in the messages. I'd love to get feedback before updating any more messages.

Changes: 
- Docs links are clickable (based on https://on.cypress.io in the text)
- Backticks added for command names, file paths, error messages (stripped in run mode)
- Blockquotes are indented (given we'd like to add code framing, didn't want to spend too much time updating these but think it'd be good to eventually add some sort of code snippet formatting)

### Examples
screen shot 2019-02-19 at 2 42 17 pm
screen shot 2019-02-19 at 2 37 19 pm
screen shot 2019-02-19 at 2 27 19 pm
screen shot 2019-02-19 at 2 22 55 pm

Before/after screenshots of all the errors I've updated in interactive and run mode can be found [here](https://airtable.com/shrQs5k3B7IHikFrX/tbl4RL7pxzADnV9Fp?blocks=hide).

File: packages/reporter/package.json [hunk 1]
```diff
@@ -19,7 +19,9 @@
     "clean-deps": "rm -rf node_modules",
     "pretest": "npm run check-deps-pre",
     "test": "node ./scripts/test.js",
-    "lint": "bin-up eslint --fix lib/*.js scripts/*.js src/*.js* src/**/*.js*"
+    "lint": "bin-up eslint --fix lib/*.js scripts/*.js src/*.js* src/**/*.js*",
+    "cypress:open": "node ../../scripts/cypress open --project .",
```
Comment (Saibamen) [inline packages/reporter/package.json]: You forgot to delete this?

**Review by Saibamen — CHANGES_REQUESTED** [review]

File: packages/reporter/package.json (hunk 1 above)
Comment (lilaconlee) [reply packages/reporter/package.json]: These were added intentionally to run the reporter Cypress tests.

File: packages/reporter/src/errors/test-error.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
+import React, { Component } from 'react'
+import Markdown from 'markdown-it'
+import FlashOnClick from '../lib/flash-on-click'
+
+const md = new Markdown('zero', {
```
Comment (jennifer-shehane) [inline packages/reporter/src/errors/test-error.jsx]: I think this is *such* a weird API they've decided on. Passing 'zero' for the base config.

**Review by jennifer-shehane — DISMISSED** [review]

**Comment by lilaconlee** [comment]
We're not going to need the link work in this PR, but it does extract the TestError into a separate component and lay some of the ground work for #3694 so I'm going to update it to remove the link clicking stuff and leave the rest.

File: packages/reporter/src/errors/test-error.jsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react'
+import FlashOnClick from '../lib/flash-on-click'
+
+class TestError extends Component {
```
Comment (chrisbreiding) [inline packages/reporter/src/errors/test-error.jsx]: Since this is stateless, I'd prefer to have it be a functional component instead of a class.

**Review by chrisbreiding — CHANGES_REQUESTED** [review]

**Review by chrisbreiding — APPROVED** [review]

**Pull request merged**